### PR TITLE
Opentec scanner

### PR DESCRIPTION
**Pull request by lamiselzobeidy**

### CODE REVIEW

File: projects/crud/src/lib/components/model-form/model-form.component.ts [hunk 1]
```diff
@@ -161,13 +162,17 @@ export class ModelFormComponent implements OnInit, OnDestroy {
...
-                this.requestAction(link);
-                break;
+        if (link.action == 'iframe') {
+            this.openIframe(link);
+        }
+        else if (link.action == 'request' && link.type == 'scan') {
+            this.fillFormControls(link);
+        }
+        else if (link.action == 'dialog') {
+            this.openOpentecPopup(link);
```
Comment (ALaaDawood) [inline projects/crud/src/lib/components/model-form/model-form.component.ts:172]: @lamiselzobeidy please make this more generalized, if the type is dialog we can call a function called open action dialog, and generalize the dialog itself

File: projects/crud/src/lib/components/model-form/model-form.component.ts [hunk 2]
```diff
@@ -181,41 +186,45 @@ export class ModelFormComponent implements OnInit, OnDestroy {
...
+                link['function'] = options.apiOptions.find(opt => opt.name == result)?.function;
+                this.fillFormControls(link);
             }
-        })
-        return res;
+        });
     }
+
+    fillFormControls(link) {
+        const linkData = link.function();
```
Comment (ALaaDawood) [inline projects/crud/src/lib/components/model-form/model-form.component.ts:208]: @lamiselzobeidy let's call `function`  a more descriptive name please

File: projects/crud/src/lib/components/model-form/model-form.component.ts [hunk 3]
```diff
@@ -181,41 +186,45 @@ export class ModelFormComponent implements OnInit, OnDestroy {
...
+        let { formSet = undefined, ...data } = { ...linkData };
+        if (formSet) {
+            for (let key in formSet) {
+                const formValue = this.formGroup.value[key];
+                let isMain = formValue.some(f => {
+                    if (f['contact_type']) {
+                        return f['contact_type']['id'] == 1;
```
Comment (ALaaDawood) [inline projects/crud/src/lib/components/model-form/model-form.component.ts:215]: @lamiselzobeidy this is too specific to user profile

File: projects/crud/src/lib/components/open-tec-popup/open-tec-popup.component.html [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<div class="container" id="confirmation-dialog">
+  <div mat-dialog-content>Would you like to capture an Identity Card or a Passport?</div>
+  <div class="actions">
+    <button *ngFor="let option of options" mat-button id="scan-passport-btn" (click)="dialogAction(option)">{{ option |
+      translate }}</button>
+  </div>
+</div>
```
Comment (ALaaDawood) [inline projects/crud/src/lib/components/open-tec-popup/open-tec-popup.component.html:7]: @lamiselzobeidy the message and the buttons need to be generic and passed to the dialog

File: projects/crud/src/lib/components/model-form/model-form.component.ts [hunk 5]
```diff
@@ -181,41 +186,43 @@ export class ModelFormComponent implements OnInit, OnDestroy {
...
-                        address: json.address1,
-                        city: json.city,
-                        zip_code: json.zip,
-                        contact_type: 1,
-                    },
-                    ...formValue
-                ]
+
+    openActionDialog(options) {
+        const dialogRef = this.dialog.open(ScannerComponent, {
```
Comment (ALaaDawood) [inline projects/crud/src/lib/components/model-form/model-form.component.ts:191]: @lamiselzobeidy let's call it actionDialog instead of ScannerComponent

**Review by ALaaDawood — CHANGES_REQUESTED** [review]
- dialog component name too specific.
- the part about formset is too specif to user profile

**Review by Shimaa-Ibrahim — APPROVED** [review]

**Review by ALaaDawood — APPROVED** [review]

**Pull request merged**